lib/api/apiUtils/object/versioning.js

```diff
@@ -434,6 +434,43 @@ function preprocessingVersioningDelete(bucketName, bucketMD, objectMD, reqVersio
     return options;
 }
 
+/**
+ * Keep metadatas when the object is restored from cold storage
+ * but remove the specific ones we don't want to keep
+ * @param {object} objMD - obj metadata
+ * @param {object} metadataStoreParams - custom built object containing resource details.
+ * @return {undefined}
+ */
+function restoreMetadatas(objMD, metadataStoreParams) {
+    /* eslint-disable no-param-reassign */
+    const userMDNotToKeep = ['x-amz-meta-scal-s3-restore-attempt'];
+    // We need to keep user metadata and tags
+    Object.keys(objMD).forEach(key => {
+        if (key.startsWith('x-amz-meta-')  && !userMDNotToKeep.includes(key)) {
+            metadataStoreParams.metaHeaders[key] = objMD[key];
+        }
+        if (key.startsWith('content-') && !userMDNotToKeep.includes(key)) {
+            metadataStoreParams[key] = objMD[key];
+        }
+    });
+
+    if (objMD.replicationInfo?.status === 'COMPLETED') {
```

Comment:
shouldn't we copy the replication status even in the other cases?

* `PENDING` should probably not happen (as data would not have been available)... though it may be that we keep retrying or just failed but did not yet update the status → maybe we can print a warning, but we should not affect this workflow
* `FAILED` can definitely happen, and we should not make it look like anyhing changed...